### ISSUE TITLE
Brownout during wifi startup

### ISSUE THREAD

**Issue by tehKaiN** [espressif/esp-idf#1308]
I'm developing ESP32-based module for board which has 500mA 3v3 power supply.

After using code attached below, each time during WiFi startup I get brownout. On scope there is quite big voltage drop. This doesn't happen on a breadboard with larger capacitor attached (470uF), so you could assume that's the problem, however AT firmware from https://github.com/espressif/esp32-at works flawlessly under same electrical conditions and starts WiFi without any problems. Unfortunately this firmware is closed source, so my question is: what can I do differently to reduce init voltage drop to have same results as in AT firmware?

I've tried `esp_wifi_set_max_tx_power` but it works only after `esp_wifi_start()` but then it's too late to make any difference.

I was suggested to post this issue here from https://github.com/espressif/arduino-esp32/issues/863

Code used:

```cpp

#include <Arduino.h>
#include "esp_wifi.h"

#define SSID "ESP32AP"

void setup() {
	wifi_init_config_t wifiInitializationConfig = WIFI_INIT_CONFIG_DEFAULT();

	esp_wifi_init(&wifiInitializationConfig);

	esp_wifi_set_storage(WIFI_STORAGE_RAM);

	esp_wifi_set_mode(WIFI_MODE_AP);

	wifi_config_t ap_config;
	strcpy((char *)ap_config.ap.ssid, SSID);
	ap_config.ap.channel = 0;
	ap_config.ap.authmode = WIFI_AUTH_OPEN;
	ap_config.ap.ssid_hidden = 0;
	ap_config.ap.max_connection = 1;
	ap_config.ap.beacon_interval = 100;

	esp_wifi_set_config(WIFI_IF_AP, &ap_config);

	esp_wifi_start();
}

void loop()
{
}
```

And then I get:

```
ets Jun  8 2016 00:22:57
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1                         
load:0x3fff0010,len:4                     
load:0x3fff0014,len:588                    
load:0x40078000,len:0                
load:0x40078000,len:9768                          
entry 0x400789b8
```

**Comment by projectgus** [comment]
Depending on what version of IDF the AT firmware & your other test firmware is compiled with, there was a bug fixed in 27b52dfd18dbc489b012848fef549916e4f58272 which reduces the peak RF calibration current. I believe Arduino-ESP32 was only updated to this IDF version last night (espressif/arduino-esp32@81a9c45a1ef5833e80c9235ff393a7641c669a94) so if your test firmware was using Arduino then it would have had a higher calibration current.

One other difference which may be significant is that the AT firmware runs at 80MHz, which has lower overall power consumption than the higher CPU clock speeds.

Despite the bug situation, I would encourage you to check if your module design has enough capacitance and add more if it does not.

**Comment by projectgus** [comment]
Hi @tehKaiN , I'm going to close this issue but please reopen if you think the above explanation doesn't cover the brownout you were seeing.

Angus
